include/realizations/catchment/Bmi_Multi_Formulation.hpp

```diff
@@ -596,6 +596,32 @@ namespace realization {
             return module->get_var_value_as_double(var_name);
         }
 
+        /**
+         * Get value vector for some BMI model variable.
+         */
+        std::vector<double> get_var_vec_as_double(const int& index, const std::string& var_name) {
+            auto data_provider_iter = availableData.find(var_name);
+            if (data_provider_iter == availableData.end()) {
+                throw external::ExternalIntegrationException(
+                        "Multi BMI formulation can't find correct nested module for BMI variable " + var_name + SOURCE_LOC);
+            }
+            // Otherwise, we have a provider, and we can cast it based on the documented assumptions
+            try {
+                std::shared_ptr <data_access::GenericDataProvider> nested_module =
+                        std::dynamic_pointer_cast<data_access::GenericDataProvider>(data_provider_iter->second);
+                long nested_module_time = nested_module->get_data_start_time() + ( this->get_model_current_time() - this->get_model_start_time() );
+                auto selector = CatchmentAggrDataSelector(this->get_catchment_id(),var_name,nested_module_time,this->record_duration(),"1");
+                //TODO: After merge PR#405, try re-adding support for index
+                return nested_module->get_values(selector);
```

Comment:
Another note here, it is probably possible to get `get_var_value_as_double` to return `const &` to avoid additional copies, but I'm not 100% sure...but think about avoiding additional copies in this PR where possible.

Reply:
Regarding the first comment, I agree that it does seem out of place for that AORC code to be there. I'll remove it from this PR. On the other hand, the code is needed at present to be able to run the gridded python module. Maybe we can put that code in ngen-forcing repo and download it whenever we want to run the Bmi-Python module in ngen?

The comments about the get_var_vec_as_double are well taken. I'll simplify the implementation along the line suggested in the comment.